src/integer_mod_q/modulus_polynomial_ring_zq/get.rs

```diff
@@ -8,11 +8,120 @@
 
 //! Implementations to get information about a [`ModulusPolynomialRingZq].
 
-use crate::{integer::Z, integer_mod_q::ModulusPolynomialRingZq};
+use crate::{
+    error::MathError,
+    integer::Z,
+    integer_mod_q::{Modulus, ModulusPolynomialRingZq, Zq},
+    traits::GetCoefficient,
+    utils::index::evaluate_index,
+};
 use flint_sys::{
     fmpz::fmpz_set,
+    fmpz_mod_poly::fmpz_mod_poly_get_coeff_fmpz,
     fq::{fq_ctx_degree, fq_ctx_struct},
 };
+use std::{fmt::Display, rc::Rc};
+
+impl GetCoefficient<Zq> for ModulusPolynomialRingZq {
+    /// Returns the coefficient of a polynomial [`ModulusPolynomialRingZq`] as a [`Zq`].
+    ///
+    /// If an index is provided which exceeds the highest set coefficient, `0` is returned.
+    ///
+    /// Parameters:
+    /// - `index`: the index of the coefficient to get (has to be positive)
+    ///
+    /// Returns the coefficient as a [`Zq`], or a [`MathError`] if the provided index
+    /// is negative and therefore invalid, or it does not fit into an [`i64`].
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::traits::*;
+    /// use qfall_math::integer_mod_q::{Zq, ModulusPolynomialRingZq};
+    /// use std::str::FromStr;
+    ///
+    /// let poly = ModulusPolynomialRingZq::from_str("4  0 1 2 3 mod 17").unwrap();
+    ///
+    /// let coeff_0: Zq = poly.get_coeff(0).unwrap();
+    /// let coeff_1: Zq = poly.get_coeff(1).unwrap();
+    /// let coeff_4: Zq = poly.get_coeff(4).unwrap();
+    ///
+    /// assert_eq!(Zq::from((0, 17)), coeff_0);
+    /// assert_eq!(Zq::from((1, 17)), coeff_1);
+    /// assert_eq!(Zq::from((0, 17)), coeff_4);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds) if
+    ///     either the index is negative or it does not fit into an [`i64`].
```

Comment:
what happens if I try to access a coefficient that is not set?, i.e. I have a polynomial of degree 5 and I try to access position 10

Reply:
"If an index is provided which exceeds the highest set coefficient, `0` is returned.". This is what we wrote in other implementations too. What should I add here to make it more clear?